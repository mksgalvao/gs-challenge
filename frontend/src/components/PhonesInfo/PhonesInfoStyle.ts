import { createStyles, Theme, makeStyles } from "@material-ui/core/styles";

export const phonesInfoStyle = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      display: "flex",
      alignContent: "center",
      justifyContent: "center",
      flexGrow: 1,
      marginTop: 30,
    },
    cardWrapper: {
      maxWidth: 600,
      maxHeight: 800,
    },
    media: {
      height: 400,
      maxWidth: "100%",
      marginTop: 30,
      marginLeft: 10,
      marginRight: 10,
    },
  })
);
