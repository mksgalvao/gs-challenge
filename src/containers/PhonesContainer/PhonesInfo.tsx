import React from "react";
import Card from "@material-ui/core/Card";

export const PhonesInfo = () => {
  return (
    <div>
      <Card> Some Good</Card>
      <h1 className="bg-primary"> bla </h1>
    </div>
  );
};
