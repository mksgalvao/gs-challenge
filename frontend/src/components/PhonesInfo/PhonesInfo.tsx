import React from "react";
import { phonesInfoStyle } from "./PhonesInfoStyle";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import CardMedia from "@material-ui/core/CardMedia";
import Typography from "@material-ui/core/Typography";

export const PhonesInfo = (props: any) => {
  const product = props.location.state.phone;
  const classes = phonesInfoStyle();

  return (
    <div className={classes.root}>
      <Card className={classes.cardWrapper}>
        <Typography paragraph>{product.name}</Typography>
        <CardMedia className={classes.media} image={product.imageFileName} />
        <CardContent>
          <Typography paragraph>
            {new Intl.NumberFormat("de-DE", {
              style: "currency",
              currency: "EUR",
            }).format(product.price)}
          </Typography>
          <Typography variant="body2" color="textSecondary" component="p">
            {product.description}
          </Typography>
          <Typography variant="body2" color="textSecondary" component="p">
            {product.color}
          </Typography>

          <Typography variant="body2" color="textSecondary" component="p">
            {product.screen}
          </Typography>

          <Typography variant="body2" color="textSecondary" component="p">
            {product.ram}
          </Typography>

          <Typography variant="body2" color="textSecondary" component="p">
            {product.processor}
          </Typography>
        </CardContent>
      </Card>
    </div>
  );
};
