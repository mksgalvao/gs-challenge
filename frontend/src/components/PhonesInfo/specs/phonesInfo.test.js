import React from "react";
import enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
enzyme.configure({ adapter: new Adapter() });

import { shallow } from "enzyme";

import { PhonesList } from "../../../fixtures/phoneFixtures";
import { PhonesInfo } from "../PhonesInfo";

describe("Testing PhonesInfo", () => {
  let phonesInfo;

  const mockedPhones = { state: { phone: PhonesList } };

  beforeEach(() => {
    phonesInfo = shallow(<PhonesInfo location={mockedPhones} />);
  });

  it("phonesInfo should render", () => {
    expect(phonesInfo).toMatchSnapshot();
  });
});
