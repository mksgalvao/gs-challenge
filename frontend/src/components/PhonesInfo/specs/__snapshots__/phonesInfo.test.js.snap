// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PhonesInfo phonesInfo should render 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Card))
    className="makeStyles-cardWrapper-2"
  >
    <WithStyles(ForwardRef(Typography))
      paragraph={true}
    />
    <WithStyles(ForwardRef(CardMedia))
      className="makeStyles-media-3"
    />
    <WithStyles(ForwardRef(CardContent))>
      <WithStyles(ForwardRef(Typography))
        paragraph={true}
      >
        NaN €
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        color="textSecondary"
        component="p"
        variant="body2"
      />
      <WithStyles(ForwardRef(Typography))
        color="textSecondary"
        component="p"
        variant="body2"
      />
      <WithStyles(ForwardRef(Typography))
        color="textSecondary"
        component="p"
        variant="body2"
      />
      <WithStyles(ForwardRef(Typography))
        color="textSecondary"
        component="p"
        variant="body2"
      />
      <WithStyles(ForwardRef(Typography))
        color="textSecondary"
        component="p"
        variant="body2"
      />
    </WithStyles(ForwardRef(CardContent))>
  </WithStyles(ForwardRef(Card))>
</div>
`;
